pkg/controller/upgradeconfig/upgradeconfig_controller_test.go

```diff
@@ -330,16 +349,18 @@ var _ = Describe("UpgradeConfigController", func() {
 
 						It("reacts accordingly", func() {
 							util.ExpectGetClusterVersion(mockKubeClient, clusterVersionList, nil)
-							gomock.InOrder(
-								mockKubeClient.EXPECT().Get(gomock.Any(), upgradeConfigName, gomock.Any()).SetArg(2, *upgradeConfig).Times(1),
-								mockValidationBuilder.EXPECT().NewClient().Return(mockValidator, nil),
-								mockValidator.EXPECT().IsValidUpgradeConfig(gomock.Any(), gomock.Any(), gomock.Any()).Return(true, nil),
-								mockMetricsClient.EXPECT().UpdateMetricValidationSucceeded(gomock.Any()),
-								mockClusterUpgraderBuilder.EXPECT().NewClient(gomock.Any(), gomock.Any(), upgradeConfig.Spec.Type).Return(mockClusterUpgrader, nil),
-								mockClusterUpgrader.EXPECT().UpgradeCluster(gomock.Any(), gomock.Any()).Times(1).Return(fakeError),
-							)
+							mockKubeClient.EXPECT().Get(gomock.Any(), upgradeConfigName, gomock.Any()).SetArg(2, *upgradeConfig).Times(1)
+							mockKubeClient.EXPECT().Status().Return(mockUpdater).Times(3)
+							mockUpdater.EXPECT().Update(gomock.Any(), gomock.Any()).Times(3)
+							mockValidationBuilder.EXPECT().NewClient().Return(mockValidator, nil)
+							mockValidator.EXPECT().IsValidUpgradeConfig(gomock.Any(), gomock.Any(), gomock.Any()).Return(true, nil)
+							mockMetricsClient.EXPECT().UpdateMetricValidationSucceeded(gomock.Any())
+							mockClusterUpgraderBuilder.EXPECT().NewClient(gomock.Any(), gomock.Any(), upgradeConfig.Spec.Type).Return(mockClusterUpgrader, nil)
+							mockClusterUpgrader.EXPECT().UpgradeCluster(gomock.Any(), gomock.Any()).Times(1).Return(upgradev1alpha1.UpgradePhaseUpgrading, &upgradev1alpha1.UpgradeCondition{}, fakeError)
+							mockKubeClient.EXPECT().Status().AnyTimes().Return(mockUpdater)
+							mockUpdater.EXPECT().Update(gomock.Any(), gomock.Any()).AnyTimes()
 							result, err := reconciler.Reconcile(reconcile.Request{NamespacedName: upgradeConfigName})
-							Expect(err).NotTo(HaveOccurred())
+							Expect(err).To(HaveOccurred())
```

Comment:
These are from:
https://github.com/openshift/managed-upgrade-operator/blob/4fd98b3f57d4eaadddc0158e336147509ccf86b8/pkg/controller/upgradeconfig/upgradeconfig_controller.go#L169-L172
we were just logging the error